src/node/desktop/src/renderer/desktop-info-bridge.ts

```diff
@@ -13,78 +13,19 @@
  *
  */
 
+import { ipcRenderer } from 'electron';
+
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
 export function getDesktopInfoBridge() {
-  if (process.platform === 'win32') {
-    return {
-      platform: '',
-      version: '',
-      scrollingCompensationType: '',
-      fixedWidthFontList: `BIZ UDGothic
-BIZ UDMincho Medium
-Cascadia Code
-Cascadia Code ExtraLight
-Cascadia Code Light
-Cascadia Code SemiBold
-Cascadia Code SemiLight
-Cascadia Mono
-Cascadia Mono ExtraLight
-Cascadia Mono Light
-Cascadia Mono SemiBold
-Cascadia Mono SemiLight
-Consolas
-Courier
-Courier New
-Lucida Console
-Lucida Sans Typewriter
-MingLiU-ExtB
-MingLiU_HKSCS-ExtB
-MS Gothic
-MS Mincho
-NSimSun
-OCR A Extended
-SimSun
-SimSun-ExtB
-UD Digi Kyokasho N-B
-UD Digi Kyokasho N-R`,
-      fixedWidthFont: 'Lucida Console',
-      proportionalFont: 'Segoe UI',
-      desktopSynctexViewer: '',
-      zoomLevel: 1.0,
-      chromiumDevtoolsPort: 0,
-    };
-  } else {
-    return {
-      platform: '',
-      version: '',
-      scrollingCompensationType: '',
-      fixedWidthFontList: `AndaleMono
-AppleBraille-Outline6Dot
-AppleBraille-Outline8Dot
-AppleBraille-Pinpoint6Dot
-AppleBraille-Pinpoint8Dot
-AppleBraille
-AppleColorEmoji
-Courier
-Courier-Oblique
-Courier-Bold
-Courier-BoldOblique
-CourierNewPSMT
-CourierNewPS-ItalicMT
-CourierNewPS-BoldMT
-CourierNewPS-BoldItalicMT
-Menlo-Regular
-Menlo-Italic
-Menlo-Bold
-Menlo-BoldItalic
-Monaco
-PTMono-Regular
-PTMono-Bold`,
-      fixedWidthFont: 'Monaco',
-      proportionalFont: 'Lucida Grande',
-      desktopSynctexViewer: '',
-      zoomLevel: 1.0,
-      chromiumDevtoolsPort: 0,
-    };
-  }
+  return {
+    platform: '',
+    version: '',
+    scrollingCompensationType: '',
+    fixedWidthFontList: ipcRenderer.sendSync('desktop_get_fixed_width_font_list'),
+    fixedWidthFont: ipcRenderer.sendSync('desktop_get_fixed_width_font'),
+    proportionalFont: ipcRenderer.sendSync('desktop_get_proportional_font'),
```

Comment:
One thing to be aware of -- historically, users (especially on Linux) have found that listing fonts can be very slow when there are a large number of fonts installed on the system. We tried to dodge that on the desktop side by making use of the font cache on Linux. See, for example:

https://github.com/rstudio/rstudio/blob/466d03b0ed61d66f3a80e7c2e530fb386818d9b4/src/cpp/desktop/DesktopInfo.cpp#L105-L146

I don't think it needs to be addressed in this PR, but I think it's worth adding an issue for QA testing later.

Reply:
Good point and I agree. I think this could be improved. The main process should asynchronously retrieve the fonts, which would be fine for the monospace font list since it is used for the appearance options. 

The proportional font needs to be discovered for the UI. Maybe we define default fonts for each platform and update the font once the system fonts have been retrieved.

For the selected fixed width font, that definitely needs changing once it reads the user's preference. There's nothing implemented so far to read the preference file that stores this setting and the file read would be an async call.

It's probably better if all of these font calls using the context bridge were changed to async. At the moment, it is awkward to call back into GWT from what I've seen. Maybe I'm not quite understanding  the correct class to use to handle promises in GWT.